java/apps/recovery-portal/src/main/webapp/password-recovery.jsp

```diff
@@ -47,8 +50,11 @@
 <%
     boolean error = IdentityManagementEndpointUtil.getBooleanValue(request.getAttribute("error"));
     String errorMsg = IdentityManagementEndpointUtil.getStringValue(request.getAttribute("errorMsg"));
-    String username = request.getParameter("username");
+    String username = StringUtils.isNotEmpty(request.getParameter("username"))
+        ? Encode.forHtmlAttribute(request.getParameter("username"))
+        : "";
     boolean isSaaSApp = Boolean.parseBoolean(request.getParameter("isSaaSApp"));
+    String sp = request.getParameter("sp");
```

Comment:
Please check whether the parameter usage is encoded properly

Reply:
Already done https://github.com/wso2/identity-apps/blob/86d6527d2e33193ca09d0cc086c28a4d10eb4528/identity-apps-core/apps/recovery-portal/src/main/webapp/password-recovery.jsp#L61